lib/ansible/modules/remote_management/manageiq/manageiq_dynamic_resource_definition.py

```diff
@@ -0,0 +1,284 @@
+      - The dynamic resource definition's name.
+  properties:
+    description:
+      - The dynamic resource definition's properties.
+    version_added: '2.6'
+'''
+
+EXAMPLES = '''
+- name: Create a new dynamic resource definition in ManageIQ
+  manageiq_dynamic_resource_definition:
+    name: 'my_custom_definition'
+    state: 'present'
+    properties:
+         attributes:
+           engine: string
+           type: string
+           size: integer
+           region: string
+           username: string
+         associations:
+           service: Service
+           provider: ManageIQ::Providers::CloudManager
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+
+- name: Edit a dynamic resource definition in ManageIQ
+  manageiq_dynamic_resource_definition:
+    name: 'my_edited_custom_definition'
+    state: 'present'
+    properties:
+         attributes:
+           group: string
+           engine: string
+           type: string
+           size: integer
+           region: string
+           username: string
+         associations:
+           service: Service
+           provider: ManageIQ::Providers::CloudManager
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+
+- name: Delete a dynamic resource definition in ManageIQ
+  manageiq_dynamic_resource_definition:
+    name: 'my_edited_custom_definition'
+    state: 'absent'
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.manageiq import ManageIQ, manageiq_argument_spec
+
+
+class ManageIQDynamicResourceDefinition(object):
+    """
+        Object to create dynamic resource definition operations in manageiq.
+    """
+
+    def __init__(self, manageiq):
+        self.manageiq = manageiq
+        self.module = self.manageiq.module
+        self.api_url = self.manageiq.api_url
+        self.client = self.manageiq.client
+
+    def dynamic_resource_definition(self, name):
+        """ Search for dynamic resource definition by name name.
+
+        Returns:
+            the dynamic resource definition, or None if not found.
+        """
+        return self.manageiq.find_collection_resource_by('generic_object_definitions', name=name)
+
+    def create_definition(self, dynamic_resource_definition, name, properties):
+        """ Creates the dynamic resource definition in manageiq.
+
+        Returns:
+            the name, created_at, updated_at, id, href, and properties.
+        """
+        # check for required arguments
+        for key, value in dict(name=name, properties=properties).items():
+            if value in (None, ''):
+                self.module.fail_json(msg="missing required argument: %s" % (key))
```

Comment:
I think you could make these required=True in module args spec, and here just assume they're set?

Reply:
Makes sense, will change it.

Reply:
Actually after reviewing it, We need to check for properties here, since it is not required if the state is absent but it is required if the state is present. I'll remove the name check since it is required for all states

Reply:
Perhaps `required_if` can help.

Reply:
:+1:  Didn't know about it. Added it.